crates/runtime/src/auth.rs

```diff
@@ -92,8 +62,31 @@ impl AuthProviders {
             .read_to_string(&mut auth_contents)
             .context(UnableToReadAuthFileSnafu)?;
 
-        self.auth =
-            toml::from_str::<AuthConfig>(&auth_contents).context(UnableToParseAuthFileSnafu)?;
+        self.auth_configs =
+            toml::from_str::<AuthConfigs>(&auth_contents).context(UnableToParseAuthFileSnafu)?;
         Ok(())
     }
 }
+
+#[allow(clippy::module_name_repetitions)]
+pub struct AuthProvider {
+    pub auth_config: AuthConfig,
+}
+
+impl AuthProvider {
+    #[must_use]
+    pub fn new(auth_config: AuthConfig) -> Self
+    where
+        Self: Sized,
+    {
+        AuthProvider { auth_config }
+    }
+
+    #[must_use]
+    pub fn get_param(&self, param: &str) -> Option<&str> {
+        self.auth_config
+            .params
+            .get(&param.to_string())
+            .map(std::string::String::as_str)
```

Comment:
nit: It's a bit ugly to have to qualify it so much - I'd prefer to put a `use std::string` and then only qualify the first part: `String::as_str`

Reply:
Apparently this works without `use std::string`!